tests/Contrib/Unit/OTLPHttpExporterTest.php

```diff
@@ -0,0 +1,214 @@
+    {
+        $client = $this->createMock(ClientInterface::class);
+        $client->method('sendRequest')->willReturn(
+            new Response($responseStatus)
+        );
+        /** @var ClientInterface $client */
+        $exporter = new Exporter($client, new HttpFactory(), new HttpFactory());
+
+        $this->assertEquals(
+            $expected,
+            $exporter->export([new Span('test.otlp.span', SpanContext::generate())])
+        );
+    }
+
+    public function exporterResponseStatusesDataProvider()
+    {
+        return [
+            'ok'                => [200, Exporter::SUCCESS],
+            'not found'         => [404, Exporter::FAILED_NOT_RETRYABLE],
+            'not authorized'    => [401, Exporter::FAILED_NOT_RETRYABLE],
+            'bad request'       => [402, Exporter::FAILED_NOT_RETRYABLE],
+            'too many requests' => [429, Exporter::FAILED_NOT_RETRYABLE],
+            'server error'      => [500, Exporter::FAILED_RETRYABLE],
+            'timeout'           => [503, Exporter::FAILED_RETRYABLE],
+            'bad gateway'       => [502, Exporter::FAILED_RETRYABLE],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider clientExceptionsShouldDecideReturnCodeDataProvider
+     */
+    public function clientExceptionsShouldDecideReturnCode($exception, $expected)
+    {
+        $client = $this->createMock(ClientInterface::class);
+        $client->method('sendRequest')->willThrowException($exception);
+
+        /** @var ClientInterface $client */
+        $exporter = new Exporter($client, new HttpFactory(), new HttpFactory());
+
+        $this->assertEquals(
+            $expected,
+            $exporter->export([new Span('test.otlp.span', SpanContext::generate())])
+        );
+    }
+
+    public function clientExceptionsShouldDecideReturnCodeDataProvider()
+    {
+        return [
+            'client'    => [
+                $this->createMock(ClientExceptionInterface::class),
+                Exporter::FAILED_RETRYABLE,
+            ],
+            'network'   => [
+                $this->createMock(NetworkExceptionInterface::class),
+                Exporter::FAILED_RETRYABLE,
+            ],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider processHeadersDataHandler
+     */
+    public function testProcessHeaders($input, $expected)
+    {
+        $headers = (new Exporter(new Client(), new HttpFactory(), new HttpFactory()))->processHeaders($input);
+
+        $this->assertEquals($expected, $headers);
+    }
+
+    public function processHeadersDataHandler()
+    {
+        return [
+            'None' => ['none', []],
+            'No Headers' => ['', []],
+            'One Header' => ['header-1=one', ['header-1' => 'one']],
+            'Two Headers' => ['header-1=one,header-2=two', ['header-1' => 'one', 'header-2' => 'two']],
+            'Invalid Headers' => ['header-1,header-2=bar,,,', ['header-2' => 'bar']],
```

Comment:
Should an invalid header return an ArgumentException or in this case silently fail?

Reply:
I'd think we would want it to return an `ArgumentException`

Reply:
I'll get that looked at. I think this would also change where the headers are processed, currently it's in the `export`. Which it should likely be done in the constructor to make it fail early.